plugins/outputs/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,238 @@
+package opentelemetry
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/pkg/errors"
+
+	metricspb "go.opentelemetry.io/proto/otlp/metrics/v1"
+	"google.golang.org/grpc/metadata"
+)
+
+// OpenTelemetry is the OpenTelemetry Protocol config info.
+type OpenTelemetry struct {
+	Endpoint    string            `toml:"endpoint"`
+	Timeout     config.Duration   `toml:"timeout"`
+	Compression string            `toml:"compression"`
+	Headers     map[string]string `toml:"headers"`
+	Attributes  map[string]string `toml:"attributes"`
+	Log         telegraf.Logger   `toml:"-"`
+	tls.ClientConfig
+
+	client       *client
+	resourceTags []*telegraf.Tag
+}
+
+const (
+	// maxInt is the max int64 value.
+	maxInt = int64(^uint(0) >> 1)
+
+	defaultEndpoint            = "http://localhost:4317"
+	defaultTimeout             = time.Second * 10
+	defaultCompression         = "gzip"
+	instrumentationLibraryName = "telegraf"
+)
+
+var sampleConfig = `
+  ## OpenTelemetry endpoint
+  # endpoint = "http://localhost:4317"
+
+  ## Timeout when sending data over grpc
+  # timeout = "10s"
+
+  ## Compression used to send data, supports: "gzip", "none"
+  # compression = "gzip"
+
+  ## Optional TLS Config for use on gRPC connections.
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+
+  # Additional resource attributes
+  [outputs.opentelemetry.attributes]
+    "service.name" = "demo"
+
+  # Additional grpc metadata
+  [outputs.opentelemetry.headers]
+    key1 = "value1"
+
+`
+
+func (o *OpenTelemetry) Init() error {
+	if o.Endpoint == "" {
+		o.Endpoint = defaultEndpoint
+	}
+	endpoint, err := url.Parse(o.Endpoint)
+	if err != nil {
+		return errors.Wrap(err, "invalid endpoint configured")
+	}
+
+	if o.Timeout < config.Duration(time.Second) {
+		o.Timeout = config.Duration(defaultTimeout)
+	}
+
+	if o.Compression == "" {
+		o.Compression = defaultCompression
+	}
+
+	for k, v := range o.Attributes {
+		o.resourceTags = append(o.resourceTags, &telegraf.Tag{Key: k, Value: v})
+	}
+
+	if o.Headers == nil {
+		o.Headers = make(map[string]string, 1)
+	}
+
+	o.Headers["telemetry-reporting-agent"] = fmt.Sprintf(
+		"%s/%s",
+		instrumentationLibraryName,
+		internal.Version(),
+	)
+
+	tlsConfig, err := o.TLSConfig()
+	if err != nil {
+		return errors.Wrap(err, "invalid tls configuration")
+	}
+
+	if o.client == nil {
```

Comment:
This is always the case as `Init()` is called first and only once, so you can skip the `if`.

Reply:
Done.